mobilecoind/src/service.rs

```diff
@@ -1927,6 +1950,26 @@ impl<T: BlockchainConnection + UserTxConnection + 'static, FPR: FogPubkeyResolve
         }
         let local_block_index = num_blocks - 1;
 
+        // Get LastBlockInfo from our peers
+        let block_infos = self.get_last_block_infos();
+        // choose the block info which is latest in terms of block index (we may be
+        // isolated from some of the nodes)
+        let last_block_info = if let Some(latest) =
+            block_infos.into_iter().reduce(|a, b| -> BlockInfo {
```

Comment:
I think `block_infos.into_iter().map(|block_info| block_info.block_index).max()` is more readable